src/platform/nrfconnect/BLEManagerImpl.cpp

```diff
@@ -20,100 +20,581 @@
  *          Provides an implementation of the BLEManager singleton object
  *          for Zephyr platforms.
  */
+
 #include <platform/internal/CHIPDeviceLayerInternal.h>
 
 #include <ble/CHIPBleServiceData.h>
 #include <platform/internal/BLEManager.h>
-
+#include <support/CodeUtils.h>
 #include <support/logging/CHIPLogging.h>
 
 #if CHIP_DEVICE_CONFIG_ENABLE_CHIPOBLE
 
+#include <bluetooth/gatt.h>
```

Comment:
Given this implementation relies on Zephyr APIs rather than something more specific to nRFConnect (e.g. the legacy SoftDevice APIs ), we should consider making moving this implementation into a more suitable directory.

Reply:
good idea

Reply:
@e-rk ?

Reply:
That can be done. I'll put it into the `Zephyr` directory where thread stack manager resides.